opentech/static_src/src/app/src/redux/actions/submissions.js

```diff
@@ -33,16 +39,67 @@ export const setCurrentSubmission = id => ({
     id,
 });
 
+
 export const loadCurrentRound = (requiredFields=[]) => (dispatch, getState) => {
-    const round = getCurrentRound(getState())
+    const state = getState()
+    const round = getCurrentRound(state)
 
-    if (round && requiredFields.every(key => round.hasOwnProperty(key))) {
+    if ( round && requiredFields.every(key => round.hasOwnProperty(key)) ) {
         return null
     }
 
-    return dispatch(fetchSubmissionsByRound(getCurrentRoundID(getState())))
+    return dispatch(fetchRound(getCurrentRoundID(state)))
 }
 
+export const loadCurrentRoundSubmissions = () => (dispatch, getState) => {
+    const state = getState()
+    const submissions = getCurrentRoundSubmissionIDs(state)
+
+    if ( submissions && submissions.length !== 0 ) {
+        return null
+    }
+
+    return dispatch(fetchSubmissionsByRound(getCurrentRoundID(state)))
+}
+
+
+export const fetchRound = roundID => {
+    return async function(dispatch) {
+        dispatch(startLoadingRound(roundID));
+        try {
+            const response = await api.fetchRound(roundID);
+            const json = await response.json();
+            if (response.ok) {
+                dispatch(updateRound(roundID, json));
+            } else {
+                dispatch(failLoadingRound(json.meta.error));
```

Comment:
Won't it be `json.detail` if there's an error? Feels like we could write a generic function to handle errors instead.